### PR TITLE
feat(image): Add new <Image> component

### PR DESCRIPTION
**Pull request by damassi**
### Description

This adds a new [react-native-fast-image](https://github.com/DylanVann/react-native-fast-image) backed `<Image>` component. By default, it will on-the-fly resize via Gemini, but this can be disabled via the `performResize={false}` prop. 

#### Fallback to screen width if no `width` is provided 
```tsx
<Image
  src="https://d32dm0rphc51dk.cloudfront.net/A983VUIZusVBKy420xP3ow/normalized.jpg"
  aspectRatio={0.62}
/>
```

#### With width and height
```tsx
<Image
  src="https://d32dm0rphc51dk.cloudfront.net/A983VUIZusVBKy420xP3ow/normalized.jpg"
  width={250}
  height={400}
/>
```

#### With aspect ratio
```tsx
<Image
  src="https://d32dm0rphc51dk.cloudfront.net/A983VUIZusVBKy420xP3ow/normalized.jpg"
  width={250}
  aspectRatio={0.62}
/>
```

#### Without automatic resizing 
```tsx
<Image
  src="https://d32dm0rphc51dk.cloudfront.net/A983VUIZusVBKy420xP3ow/normalized.jpg"
  aspectRatio={0.62}
  performResize={false}
/>
```

cc @artsy/mobile-platform

### CODE REVIEW

File: src/elements/Image/Image.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  let imageHeight
+
+  if (props.height) {
+    imageHeight = props.height
+  } else {
+    if (!props.aspectRatio) {
+      console.error("[CachedImage] Error: `aspectRatio` is required if `height` is not provided.")
+    }
+
+    const aspectRatio = props.aspectRatio ?? 1
```
Comment (damassi) [inline src/elements/Image/Image.tsx:86]: Log a red screen in dev that we need something, but don't blow up the UI

**Comment by artsyit** [comment]
:rocket: PR was released in [`v11.2.0`](https://github.com/artsy/palette-mobile/releases/tag/v11.2.0) :rocket:

File: src/utils/createGeminiUrl.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { Platform } from "react-native"
+
+export function createGeminiUrl({
```
Comment (damassi) [inline src/utils/createGeminiUrl.ts:3]: just copied this from Eigen

**Review by gkartalis — COMMENTED** [review]
Looks great! Some observations for energy:

when updating `palette-mobile` energy and trying to use the `<Image />` component from palette I got this error:

```
[Invariant Violation: requireNativeComponent: "FastImageView" was not found in the UIManager in react native](https://stackoverflow.com/questions/58908710/invariant-violation-requirenativecomponent-fastimageview-was-not-found-in-th)
```

what solved it for me was to also install `react-native-fast-image` on energy and then it stopped complaining (see the PR here // add PR).

So maybe we should update the [how to install](https://github.com/artsy/palette-mobile#how-to-install) readme section to reflect that (adding react-native-fast-image)

**Comment by gkartalis** [comment]
Another thing: when trying to use the <Image /> component noticed that it was also crashing because in energy we didn't use the [ScreenDimensionProvider](https://github.com/artsy/energy/pull/654/commits/9f5d2d6b6f7af2468eb26787d7e42e40b5935dd3) that we use inside the <Image /> component so I added it and we might need to clean up ProvideScreenDimensions from energy  (I can do it on the same PR)

**Comment by damassi** [comment]
Thanks @gkartalis for giving it a spin 👍 Yeah i had the same [q about deps here](https://artsy.slack.com/archives/C02NAJ2CGLW/p1686863393013419)! With the native deps it wasn't totally clear what to do

**Pull request merged**